rm-datasource/src/main/java/io/seata/rm/datasource/undo/KeywordCheckerFactory.java

```diff
@@ -28,20 +29,31 @@
  */
 public class KeywordCheckerFactory {
 
+    private static volatile Map<String,KeywordChecker> keywordCheckerMap;
+
     /**
      * get keyword checker
      *
      * @param dbType the db type
      * @return keyword checker
      */
     public static KeywordChecker getKeywordChecker(String dbType) {
-        if (dbType.equals(JdbcConstants.MYSQL)) {
-            return MySQLKeywordChecker.getInstance();
-        } else  if (dbType.equals(JdbcConstants.ORACLE)) {
-            return OracleKeywordChecker.getInstance();
-        } else {
-            throw new NotSupportYetException(dbType);
+        if (keywordCheckerMap == null) {
+            synchronized (KeywordCheckerFactory.class) {
+                if (keywordCheckerMap == null) {
+                    Map<String, KeywordChecker> initializedMap = new HashMap<>();
+                    List<KeywordChecker> checkerList = EnhancedServiceLoader.loadAll(KeywordChecker.class);
+                    for (KeywordChecker checker : checkerList) {
+                        initializedMap.put(checker.getDbType().toLowerCase(), checker);
+                    }
+                    keywordCheckerMap = initializedMap;
+                }
+            }
+        }
+        if (keywordCheckerMap.containsKey(dbType)) {
```

Comment:
dbType. toLowerCase()?

Reply:
This is because I noticed that all database types in jdbcconstants are lowercase, and what is not sure is whether the provider of keywordchecker will report its own database types in the correct case, but it seems a little strange to do so(call toLowerCase()).
In fact, when the specific keywordchecker provides the wrong dbtype（just like POSTGRES not postgres）, to throw an exception in Factory is also a reasonable way, so I removed tower case () finally

Reply:
BaseTransactionalExecutor#getDbType，it depends on the jdbc interface to return. Is this guaranteed to return lowercase? Here I suggest lowercase or change the parameters to known enum types.